docs/kubernetes-compatibility.md

```diff
@@ -0,0 +1,53 @@
+# Kubernetes Version Compatibility
+
+Kubernetes supports 3 minor release concurrently. This means if the latest
+Kubernetes release is v1.19, that v1.18 and v0.17 will remain supported for
+backport fixes but that v0.16 will lose support.
+
+Similarly, each KubeVirt release maintains compatibility with the latest 3
+Kubernetes releases that were out at the time the KubeVirt release is made.
+
+# Compatibility Matrix Examples.
+
+Here are some theoretical examples (note that the versions are made up to
+illustrate the example)
+
+## New KubeVirt Release
+
+KubeVirt release v0.1 is cut. At that point in time the latest Kubernetes
```

Comment:
Can we rephrase that a little bit? It can very well be that CI is not ready to consume the latest Kubernetes release. Maybe we can add that it depends on what we were testing when we did a release.

Reply:
i could, but that kind of invalidates everything in this doc. The doc is committing that master will be tracking the latest Kubernetes releases.

HOw about this. I'll add a sentence saying there is some lag time between when Kubernetes release occurs and when it makes it into our CI, just so expectations are set.

Reply:
I do think that part of our commitment should be to only cut a release once it is validated on teh latest kubernetes.

Reply:
may make sense, but that has implicit impact on how we do things on CI and may delay releases in a way we didn't have before.

Reply:
any thoughts on how I might be able to loosen the commitment here? like, x number of months after a new k8s release? i'd like some timeframe, but it doesn't have to be exact. 

maybe i could say, "in general we aim to adopt new k8s releases into our CI within a month of their release" ?

Reply:
One month sounds good. I would still insist that we we add that we support the k8s versions which we tested at release time with a release. For instance 1.18 can be out for two weeks but we still only test 1.17 becaue we may have issues running the tests on 1.18.

I could see us trying to consume release candidates once we have our automation better under control and try to minimize the time window.

Reply:
makes sense to me